symphony-bdk-core/src/main/java/com/symphony/bdk/core/auth/impl/AuthSessionImpl.java

```diff
@@ -14,12 +14,12 @@
 @API(status = API.Status.INTERNAL)
 public class AuthSessionImpl implements AuthSession {
 
-  private final BotAuthenticatorRsaImpl authenticator;
+  private final BotAuthenticator authenticator;
 
   private String sessionToken;
   private String keyManagerToken;
 
-  public AuthSessionImpl(@Nonnull BotAuthenticatorRsaImpl authenticator) {
+  public AuthSessionImpl(@Nonnull BotAuthenticator authenticator) {
```

Comment:
Not really related to your PR but do you think we need to retrieve session token and keymanager token at the moment that the `AuthSession` is initiated, instead of let it null and retrieve them by calling `refresh()`?

```
public AuthSessionImpl(@Nonnull BotAuthenticatorRsaImpl authenticator) throws AuthUnauthorizedException {
    this.authenticator = authenticator;
    this.sessionToken = authenticator.retrieveSessionToken();
    this.keyManagerToken = authenticator.retrieveKeyManagerToken();
  }
```

Reply:
Maybe let us check that with @symphony-thibault

Reply:
Already implemented in https://github.com/SymphonyPlatformSolutions/symphony-api-client-java/pull/181

Reply:
@symphony-thibault What do you think ?

Reply:
Correct, let's follow the same way as already done here : https://github.com/SymphonyPlatformSolutions/symphony-api-client-java/blob/23798f6d6200f7e6fed1dac1d3d7828962f9b7ed/symphony-bdk-core/src/main/java/com/symphony/bdk/core/auth/impl/BotAuthenticatorRsaImpl.java#L55